quic-client/src/nonblocking/quic_client.rs

```diff
@@ -38,31 +39,69 @@ use {
     tokio::{sync::OnceCell, time::timeout},
 };
 
+#[derive(Debug)]
 pub struct SkipServerVerification;
 
 impl SkipServerVerification {
-    pub fn new() -> Arc<Self> {
+    fn new() -> Arc<Self> {
         Arc::new(Self)
     }
 }
 
-impl rustls::client::ServerCertVerifier for SkipServerVerification {
+impl rustls::client::danger::ServerCertVerifier for SkipServerVerification {
+    fn verify_tls12_signature(
+        &self,
+        _message: &[u8],
+        _cert: &rustls::pki_types::CertificateDer<'_>,
+        _dss: &rustls::DigitallySignedStruct,
+    ) -> Result<rustls::client::danger::HandshakeSignatureValid, rustls::Error> {
+        Ok(rustls::client::danger::HandshakeSignatureValid::assertion())
+    }
+
+    fn verify_tls13_signature(
+        &self,
+        _message: &[u8],
+        _cert: &rustls::pki_types::CertificateDer<'_>,
+        _dss: &rustls::DigitallySignedStruct,
+    ) -> Result<rustls::client::danger::HandshakeSignatureValid, rustls::Error> {
+        Ok(rustls::client::danger::HandshakeSignatureValid::assertion())
```

Comment:
Not checking the signature here would allow clients to impersonate any staked node. 
Signature verification should happen here, but it's probably worth implementing a custom verifier using dalek instead of using the ring crypto backend to improve handshake performance.

Reply:
On second thought, I think Agave is using the source IP address to enforce stake weight QoS, not the identity key, so this attack wouldn't work.

Reply:
Changed to use rustls::crypto::CryptoProvider to handle this.